File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """

        buy = prices[0]
        p = 0

        for i in range(1, len(prices)):
            cost = prices[i] - buy
            # if(prices[i] - buy) > 0:
            #     p = max(p, prices[i]-buy)
            p = max(p, cost)
            buy = min(prices[i], buy)

        return p

        

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int buy = prices[0], profit = 0;

        for(int i = 1; i<n; i++){
            int cost = prices[i] - buy;
            profit = max(profit, cost);
            buy = min(buy, prices[i]);
        }

        return profit;
    }
};